packages/react-ui-core/src/Gmap/FreeDrawLayer.js

```diff
@@ -0,0 +1,92 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import autobind from 'autobind-decorator'
+import themed from 'react-themed'
+
+@themed(['FreeDraw', 'FreeDraw_ToolTip'], { pure: true })
+export default class FreeDrawLayer extends PureComponent {
+  static propTypes ={
+    map: PropTypes.object,
+    onMapDrawStart: PropTypes.func,
+    onMapDrawEnd: PropTypes.func,
+    theme: PropTypes.object,
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      mapControls: {
+        fullscreenControl: false,
+        streetViewControl: false,
+        zoomControl: true,
+        mapTypeControl: true,
+        scaleControl: true,
+        rotateControl: true,
+      },
+    }
+  }
+
+  enableMapControls() {
+    this.props.map.setOptions({
+      draggable: true,
+      zoomControl: true,
+      scrollwheel: true,
+      disableDoubleClickZoom: true,
+    })
+  }
+
+  disableMapControls() {
+    this.props.map.setOptions({
+      draggable: false,
+      zoomControl: false,
+      scrollwheel: false,
+      disableDoubleClickZoom: false,
+    })
+  }
+
+  @autobind
+  enableMapDraw() {
+    const { map } = this.props
+    window.google.maps.event.addDomListener(map.getDiv(), 'mousedown', () => {
+      this.drawFreeHand()
+    })
+  }
+
+  drawFreeHand() {
+    const { map, onMapDrawStart, onMapDrawEnd } = this.props
+    this.disableMapControls()
+    const poly = new window.google.maps.Polyline({ map, clickable: false })
+
+    const move = window.google.maps.event.addListener(map, 'mousemove', e => {
+      poly.getPath().push(e.latLng)
+      if (onMapDrawStart) {
+        onMapDrawStart()
+      }
+    })
+
+    window.google.maps.event.addListenerOnce(map, 'mouseup', () => {
+      window.google.maps.event.removeListener(move)
+      const path = poly.getPath()
+      poly.setMap(null)
+      const polygon = new window.google.maps.Polygon({ map, path })
+      window.google.maps.event.clearListeners(map.getDiv(), 'mousedown')
+      if (onMapDrawStart) {
+        onMapDrawEnd(polygon)
+      }
+      this.enableMapControls()
+    })
+  }
+
+  render() {
+    const { theme } = this.props
+
+    return (
+      <button
+        className={theme.FreeDraw_ToolTip}
+        onClick={this.enableMapDraw}
+      >
+        Click to draw
```

Comment:
Make this a prop with a default value?

Reply:
This is just a temporary thing until the tooltip is built, then there will be a default tooltip/button. Just not sure what it will look like once finished so not sure it makes sense to set that up right now